drivers/flash/flash_stm32_qspi.c

```diff
@@ -20,6 +20,17 @@
 #include <drivers/dma.h>
 #include <drivers/dma/dma_stm32.h>
 
+#if DT_INST_NODE_HAS_PROP(0, spi_bus_width) && \
+	DT_INST_PROP(0, spi_bus_width) == 4
+#define STM32_QSPI_QUAD_IO 1
```

Comment:
Redefine it as something `_ENABLED` because the `STM32_QSPI_QUAD_IO` sounds like index of some IO or number of IOs.

Reply:
Maybe it shall be `STM32_QSPI_USE_QUAD_IO` ? 
(The same approach in the same file is with `STM32_QSPI_USE_DMA`)

I would like to avoid the situation when we would have `IS_ENABLED(STM32_QSPI_QUAD_IO_ENABLED)` as it looks
like some kind of the overkill.

Reply:
Checkpatch failed to detect your usage if `IS_ENABLED` without config, this should generate warning: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/checkpatch.pl#L6511-L6515

Reply:
Ok, Then I will use `STM32_QSPI_USE_QUAD_IO` directly in the `if (` statement.

Reply:
> Checkpatch failed to detect your usage if `IS_ENABLED` without config, this should generate warning: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/checkpatch.pl#L6511-L6515

This is not the case anymore, cf https://github.com/zephyrproject-rtos/zephyr/pull/39390

Reply:
touché

Reply:
So then I shall use:
`if (IS_ENABLED(STM32_QSPI_USE_QUAD_IO) ... )` ?

Reply:
Yes, you can use it.